### PR TITLE
Set wallet properly - receive tab

### PR DESCRIPTION
**Pull request by JDonadio**
Fixes https://github.com/bitpay/copay/issues/5773

### CODE REVIEW

File: src/js/controllers/tab-receive.js [hunk 1]
```diff
@@ -127,26 +108,47 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
...
+      $scope.wallet.status = status;
+      $scope.setAddress();
     });
   };
 
   $scope.$on("$ionicView.beforeEnter", function(event, data) {
     $scope.wallets = profileService.getWallets();
+    if (!$scope.wallets[0]) return;
+    if ($scope.wallet && findWallet($scope.wallet))
+      $scope.setWallet($scope.wallets.indexOf($scope.wallet));
```
Comment (isocolsky) [inline src/js/controllers/tab-receive.js]: Since you already have `$scope.wallet` why not just use it as the argument to `setWallet()`?

Comment (isocolsky) [inline src/js/controllers/tab-receive.js]: Even better: accept a wallet object as the argument to `setWallet()`

**Comment by JDonadio** [comment]
Updated. thanks for your comments!

File: src/js/controllers/tab-receive.js [hunk 2]
```diff
@@ -88,9 +82,9 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
     });
   };
 
-  $scope.setWallet = function(index) {
-    $scope.wallet = $scope.wallets[index];
-    $scope.walletIndex = index;
+  $scope.setWallet = function(wallet) {
+    $scope.wallet = wallet;
+    $scope.walletIndex = $scope.wallets.indexOf(wallet);
     if ($scope.walletAddrs[$scope.wallet.id].addr) $scope.addr = $scope.walletAddrs[$scope.walletIndex].addr;
```
Comment (isocolsky) [inline src/js/controllers/tab-receive.js]: What would happen if `$scope.walletAddrs[$scope.wallet.id].addr` was `undefined`? I think you should reset `$scope.addr` first.

File: src/js/controllers/tab-receive.js [hunk 3]
```diff
@@ -88,9 +82,9 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
     });
   };
 
-  $scope.setWallet = function(index) {
-    $scope.wallet = $scope.wallets[index];
-    $scope.walletIndex = index;
+  $scope.setWallet = function(wallet) {
```
Comment (isocolsky) [inline src/js/controllers/tab-receive.js]: This looks better but the view is still using this function passing the wallet index: https://github.com/bitpay/copay/blob/master/www/views/tab-receive.html#L80

File: src/js/controllers/tab-receive.js (hunk 2 above)
Comment (JDonadio) [inline src/js/controllers/tab-receive.js]: $scope.addr is cleaned in `setAddress` function se branch

File: src/js/controllers/tab-receive.js (hunk 3 above)
Comment (JDonadio) [inline src/js/controllers/tab-receive.js]: You're right. Changed.

File: src/js/controllers/tab-receive.js [hunk 4]
```diff
@@ -88,9 +82,9 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
...
+    $scope.wallet = wallet;
+    $scope.walletIndex = $scope.wallets.indexOf(wallet);
```
Comment (JDonadio) [inline src/js/controllers/tab-receive.js]: Unnecesary, it's not used in the view. Chenged to `var walletIndex`

**Comment by JDonadio** [comment]
Closed in favor of https://github.com/bitpay/copay/pull/5884

**Pull request closed without merging**